server/src-lib/Hasura/RQL/Types/ScheduledTrigger.hs

```diff
@@ -0,0 +1,103 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Hasura.RQL.Types.ScheduledTrigger
+  ( ScheduleType(..)
+  , CreateScheduledTrigger(..)
+  , RetryConfST(..)
+  , formatTime'
+  ) where
+
+import           Data.Time.Clock
+import           Data.Time.Clock.Units
+import           Data.Time.Format
+import           Data.Aeson
+import           Data.Aeson.Casing
+import           Data.Aeson.TH
+import           Hasura.Prelude
+import           System.Cron.Types
+import           System.Cron.Parser
+import           Hasura.Incremental
+
+import qualified Data.Text                     as T
+import qualified Data.Aeson                    as J
+import qualified Hasura.RQL.Types.EventTrigger as ET
+
+data RetryConfST
+  = RetryConfST
+  { rcstNumRetries  :: !Int
+  , rcstIntervalSec :: !Seconds
+  , rcstTimeoutSec  :: !Seconds
+  , rcstTolerance   :: !Seconds
```

Comment:
I am not 100% sure if I did a good job of using this at the call-sites. E.g.

https://github.com/hasura/graphql-engine/pull/3553/files#diff-9935c1913dc1f1b962da141c6cc013bcR203

https://github.com/hasura/graphql-engine/pull/3553/files#diff-9935c1913dc1f1b962da141c6cc013bcR261